crates/bevy_ui/src/render/mod.rs

```diff
@@ -170,17 +181,104 @@ pub struct ExtractedUiNodes {
     pub uinodes: EntityHashMap<Entity, ExtractedUiNode>,
 }
 
-pub fn extract_atlas_uinodes(
+pub fn extract_uinode_background(
+    mut extracted_uinodes: ResMut<ExtractedUiNodes>,
+    uinode_query: Extract<
+        Query<(
+            Entity,
+            &Node,
+            &GlobalTransform,
+            &ViewVisibility,
+            Option<&CalculatedClip>,
+            &BackgroundColor,
+        )>,
+    >,
+) {
+    for (entity, uinode, transform, view_visibility, clip, background_color) in uinode_query.iter()
+    {
+        // Skip invisible backgrounds
+        if !view_visibility.get() || background_color.0.is_fully_transparent() {
+            continue;
+        }
+
+        extracted_uinodes.uinodes.insert(
+            entity,
+            ExtractedUiNode {
+                stack_index: uinode.stack_index,
+                transform: transform.compute_matrix(),
+                color: background_color.0,
+                rect: Rect {
+                    min: Vec2::ZERO,
+                    max: uinode.calculated_size,
+                },
+                clip: clip.map(|clip| clip.clip),
+                image: AssetId::default(),
+                atlas_size: None,
+                flip_x: false,
+                flip_y: false,
+            },
+        );
+    }
+}
+
+pub fn extract_uinode_image(
```

Comment:
It extracts multiple images, so should be `images` plural I think
```suggestion
pub fn extract_uinode_images(
```